draft-jmiller-json-web-proof.md

```diff
@@ -70,49 +71,25 @@ when, and only when, they appear in all capitals, as shown here.
 
 A _JSON Web Proof (JWP)_ is very similar to a JWS [@RFC7515], with the addition that it can contain multiple individual payloads instead of a singular one.  New JWP-supporting algorithms are then able to separate and act on the individual payloads contained within.
 
-In addition to the JWS `sign` and `verify` interactions, JWP also importantly adds a `derive` processing step for interaction with the algorithm to perform the selective disclosure and privacy preserving transformations.  This allows for multi-party interactions where a credential is issued from one party, derived by an intermediary party, then presented to another verifying party.  While `sign` only occurs once on a JWP, `derive` and `verify` may be repeated if supported by the algorithm.
+In addition to the JWS `sign` and `verify` interactions, JWP also importantly adds a `prove` processing step for interaction with the algorithm to perform the selective disclosure and privacy preserving transformations.  This allows for multi-party interactions where a credential is issued from one party, derived by the prooving party, then presented to another verifying party.  While `sign` only occurs once to create a JWP, the `prove` and `verify` steps may be safely repeated when supported by the algorithm.
```

Comment:
I would prefer to see JWP support interfaces like:

`create/generate` - construct and secure document and proof.
`derive/prove` - transform a document and proof in such a way that they remain verifiable while's enhancing privacy.
`verify` - verify a document and proof.

I'm not sure its wise to limit `sign/create` to happening only once.... consider multi-signature scenarios where material is collected from parties over time.

We could potential address those scenarios by thinking of `sign` as `contribute to proof` where it might happen once, or a few times.

Reply:
I mentioned in another comment my thought process behind sign/sign_verify and proove/proof_verify language, happy to pick that up in this comment thread or we can create an issue to continue also.

Good point about the "only once" language, fixing that now :+1: